lib/task/import/validate/csvRepoValidator.class.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV Repository Validator. Check if repo exists or not in the AtoM DB.
+ * Issue a warning if the CSV will result in the creation of a new Repo.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ */
+class CsvRepoValidator extends CsvBaseValidator
+{
+    const TITLE = 'Repository Check';
+    const LIMIT_TO = ['QubitInformationObject'];
+
+    protected $existingRepositories = [];
+    protected $newRepositories = [];
+    protected $repositoryColumnPresent;
+
+    public function __construct(?array $options = null)
+    {
+        $this->setTitle(self::TITLE);
+
+        parent::__construct($options);
+    }
+
+    public function reset()
+    {
+        //$this->values = [];
+        $this->repositoryColumnPresent = null;
+        $this->newRepositories = [];
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+        $row = $this->combineRow($header, $row);
+
+        // Set if repository column is present.
+        if (!isset($this->repositoryColumnPresent)) {
+            $this->repositoryColumnPresent = isset($row['repository']);
+        }
+
+        if (!$this->repositoryColumnPresent || empty($row['repository'])) {
+            return;
+        }
+
+        if (!$this->repositoryExists($row['repository'])) {
+            $this->testData->addDetail(implode(',', $row));
+        }
+    }
+
+    public function getTestResult()
+    {
+        if (!$this->repositoryColumnPresent) {
+            // repository column not present in file.
+            $this->testData->addResult(sprintf("'repository' column not present in file."));
+
+            return parent::getTestResult();
+        }
+
+        if (0 < count($this->newRepositories)) {
```

Comment:
Nitpick: These two conditionals could be combined as an if/else I think.

Reply:
Correct - I will combine these.